src/lex/mod.rs

```diff
@@ -305,20 +305,20 @@ impl Lexer {
         } else if self.match_next(b'L') {
             self.match_next(b'L');
         }
-        if radix == 2 {
+        if radix == Radix::Binary {
             let span = self.span(span_start);
             self.error_handler
                 .warn("binary number literals are an extension", span);
         }
         Ok(Token::Literal(literal))
     }
     // at this point we've already seen a '.', if we see one again it's an error
-    fn parse_float(&mut self, radix: u32, mut buf: String) -> Result<f64, LexError> {
+    fn parse_float(&mut self, radix: Radix, mut buf: String) -> Result<f64, LexError> {
         buf.push('.');
         // parse fraction: second {digits} in regex
         while let Some(c) = self.peek() {
             let c = c as char;
-            if c.is_digit(radix) {
+            if c.is_digit(radix.as_u8() as u32) {
```

Comment:
Use `into()` instead of `as` for non-lossy conversions.
```suggestion
            if c.is_digit(radix.as_u8().into()) {
```